src/backend/distributed/test/metadata_sync.c

```diff
@@ -0,0 +1,64 @@
+/*-------------------------------------------------------------------------
+ *
+ * test/src/metadata_sync.c
+ *
+ * This file contains functions to exercise the metadata snapshoy
+ * generation functionality within Citus.
+ *
+ * Copyright (c) 2014-2016, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+#include "postgres.h"
+#include "c.h"
+#include "fmgr.h"
+
+#include "catalog/pg_type.h"
+#include "distributed/metadata_sync.h"
+#include "distributed/test_helper_functions.h" /* IWYU pragma: keep */
+#include "utils/array.h"
+#include "utils/builtins.h"
+
+
+/* declarations for dynamic loading */
+PG_FUNCTION_INFO_V1(cluster_metadata_snapshot);
+
+
+/*
+ * cluster_metadata_snapshot prints all the queries that are required
+ * to generate a metadata snapshot.
+ */
+Datum
+cluster_metadata_snapshot(PG_FUNCTION_ARGS)
```

Comment:
Maybe just call it generate_metadata_snapshot, since we're sort of phasing out cluster_ as a prefix.